kernel/common/string.rb

```diff
@@ -42,8 +42,9 @@ def clone
     alias_method :dup, :clone
   end
 
-  def initialize(arg = undefined)
+  def initialize(arg = undefined, encoding: nil)
     replace arg unless undefined.equal?(arg)
+    self.force_encoding(encoding) if encoding
```

Comment:
After digging into mri's c code @YorickPeterse and me came to the conclusion to use force_encoding here.
MRI is using `rb_enc_associate`: https://github.com/ruby/ruby/blob/trunk/string.c#L1354
Which is also used in `force_encoding`: https://github.com/ruby/ruby/blob/trunk/string.c#L8671